### PR TITLE
Bug 2076193: Remove special-casing for default ingresscontroller probe parameters

### PR DESCRIPTION
**Pull request by Miciah**
Remove logic that special-cases handling of the default ingresscontroller.  This special-casing prevented the user from modifying the kubelet probe timeouts of the default ingresscontroller while allowing the user to modify the kubelet probe timeouts for custom ingresscontrollers.  However, the special-casing was not part of [the enhancement design document](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/tunable-router-kubelet-probes.md), so it should not be in the implementation.  After this change, the user can modify the kubelet probe timeouts for default and custom ingresscontrollers alike.

Follow-up to #736.

* `pkg/operator/controller/ingress/deployment.go` (`ensureRouterDeployment`): Delete the `defaultIngressController` variable and argument to `desiredRouterDeployment` and `updateRouterDeployment`.
(`desiredRouterDeployment`, `deploymentHash`, `deploymentTemplateHash`, `hashableDeployment`, `hashableProbe`, `updateRouterDeployment`, `deploymentConfigChanged`): Delete the `defaultIngressController` parameter and special-casing for the default ingresscontroller.
(`hashableProbe`): Use `copyProbe`.
(`copyProbe`): Copy logic that was previously in `hashableProbe`.  With this change, `hashableProbe` and `deploymentConfigChanged` both use `copyProbe` and thus behave consistently for all probe parameters.
* `pkg/operator/controller/ingress/deployment_test.go` (`checkDeploymentHash`): Delete the `defaultIngressController` argument to `deploymentTemplateHash`.
(`TestDesiredRouterDeployment`): Delete the `f`defaultIngressController argument to `desiredRouterDeployment`.
(`TestDeploymentHash`): Delete the `defaultIngressController` argument to `deploymentHash` and `deploymentTemplateHash`.
(`TestDeploymentConfigChanged`): Delete test cases for changing probe parameters for the default ingresscontroller as it is no longer special-cased.
* `test/e2e/operator_test.go` (`TestTunableRouterKubeletProbesForDefaultIngressController`): Delete test.

### CODE REVIEW

**Comment by frobware** [comment]
Just noting/observing the order of the RBAC cluster-ingress-operator e2e test failures: 

--- PASS: TestConfigurableRouteNoSecretNoRBAC (0.93s)
=== RUN   TestConfigurableRouteNoConsumingUserNoRBAC
    configurable_route_test.go:421: error updating ingress.status: Operation cannot be fulfilled on ingresses.config.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again
    configurable_route_test.go:301: failed to restore cluster ingress resource to original state: Operation cannot be fulfilled on ingresses.config.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again
    configurable_route_test.go:305: failed to restore cluster ingress resource to original state: Operation cannot be fulfilled on ingresses.config.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again
--- FAIL: TestConfigurableRouteNoConsumingUserNoRBAC (2.06s)

File: pkg/operator/controller/ingress/deployment.go [hunk 1]
```diff
@@ -1486,17 +1459,33 @@ func deploymentConfigChanged(current, expected *appsv1.Deployment, defaultIngres
...
+			Path: a.ProbeHandler.HTTPGet.Path,
+			Port: a.ProbeHandler.HTTPGet.Port,
+			Host: a.ProbeHandler.HTTPGet.Host,
+		}
+		if a.ProbeHandler.HTTPGet.Scheme != "HTTP" {
+			b.ProbeHandler.HTTPGet.Scheme = a.ProbeHandler.HTTPGet.Scheme
+		}
+	}
+
+	// Users are permitted to modify the timeout, so *don't* copy it.
```
Comment (frobware) [inline pkg/operator/controller/ingress/deployment.go:1478]: Is changing the startupProbe timeout intentionally permitted?

```sh
$ oc -n openshift-ingress patch deploy/router-default --type=strategic --patch='{"spec":{"template":{"spec":{"$setElementOrder/containers":[{"name":"router"}],"containers":[{"livenessProbe":{"timeoutSeconds":1101},"name":"router","readinessProbe":{"timeoutSeconds":1202},"startupProbe":{"timeoutSeconds":1203}}]}}}}'
Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "router" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "router" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "router" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "router" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
deployment.apps/router-default patched
```

Wait a while....

```sh
$ sleep 30
$ oc -n openshift-ingress get  deploy/router-default -o yaml | grep -A8 Probe:
        livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /healthz
            port: 1936
            scheme: HTTP
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1101
--
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /healthz/ready
            port: 1936
            scheme: HTTP
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1202
--
        startupProbe:
          failureThreshold: 120
          httpGet:
            path: /healthz/ready
            port: 1936
            scheme: HTTP
          periodSeconds: 1
          successThreshold: 1
          timeoutSeconds: 1203
```

**Review by frobware — COMMENTED** [review]
LGTM but just wanted clarification whether changing the startupProbe timeout was intentionally permitted.

**Comment by frobware** [comment]
> Just noting/observing the order of the RBAC cluster-ingress-operator e2e test failures:
> 
> --- PASS: TestConfigurableRouteNoSecretNoRBAC (0.93s) === RUN TestConfigurableRouteNoConsumingUserNoRBAC configurable_route_test.go:421: error updating ingress.status: Operation cannot be fulfilled on ingresses.config.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again configurable_route_test.go:301: failed to restore cluster ingress resource to original state: Operation cannot be fulfilled on ingresses.config.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again configurable_route_test.go:305: failed to restore cluster ingress resource to original state: Operation cannot be fulfilled on ingresses.config.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again --- FAIL: TestConfigurableRouteNoConsumingUserNoRBAC (2.06s)

Attempting to fix this with PR #744.

File: pkg/operator/controller/ingress/deployment.go (hunk 1 above)
Comment (Miciah) [reply pkg/operator/controller/ingress/deployment.go:1478]: I intentionally wrote the code that way, but I didn't mention it in [the enhancement document](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/tunable-router-kubelet-probes.md), and [RFE-1786](https://issues.redhat.com/browse/RFE-1796) doesn't request the capability.  I don't feel strongly about this; would you prefer I remove the capability to modify the startup probe timeout?

Comment (frobware) [reply pkg/operator/controller/ingress/deployment.go:1478]: No. Though it is an additional thing for us to check when debugging live issues.

/lgtm

**Comment by Miciah** [comment]
Rebased for #585.

**Pull request merged**